code/src/main/orxAndroidNativeSupport.cpp

```diff
@@ -470,6 +731,36 @@ void android_main( android_app* state )
         orxAndroid_JNI_SetupThread(orxNULL);
     }
 
+    JNIEnv *env = Android_JNI_GetEnv();
+
+    jobject jActivity = state->activity->clazz;
+    sstAndroid.mActivity = env->NewGlobalRef(jActivity);

```

Comment:
> not a good idea to keep a global ref here,
> you are leaking the activity

Well spotted!

Reply:
Ok, I copied this pattern from the other "android" glue code orxAndroidSupport.cpp without reflecting over storing activity instance that can destroyed.  So you mean that the state->activity->clazz can be updated if a new instance of the Activity is created?
Anyway I'll revert this so the same code as before (using state->activity->clazz).

For that other "android" case this isn't a problem I think  because the OrxActivity.onDestroy() calls stopOrx() in glue code which sends the orxSYSTEM_EVENT_CLOSE to Orx but the Java Thread instance mOrxThread in the Activity is not stopped from what I can find. And the sstAndroid.mActivity instance in glue code lives in that thread. Could it be that we have the same problem there then?

Reply:
in the non native AndroidSupport, there is a env->DeleteGlobalRef(sstAndroid.mActivity) in orxAndroid_Exit

Reply:
Yes, I see now.